src/DynamoCore/Library/XmlDocumentationExtensions.cs

```diff
@@ -199,6 +222,81 @@ private static string GetMemberElementName(FunctionDescriptor member)
             return String.Format("{0}:{1}", prefixCode, memberName);
         }
 
+        private enum XmlTagType
+        {
+            None,
+            Member,
+            Summary,
+            Parameter,
+            SearchTags
+        }
+
+        private static void LoadDataFromXml(XmlReader reader)
+        {
+            if (reader == null)
+                return;
+
+            MemberDocumentNode currentDocNode = new MemberDocumentNode();
```

Comment:
Why do we initialize `MemberDocumentNode currentDocNode` here, only to re-initialize it again below? Is that because `member` does not always come before other attributes? If that's the case, can we implement a new helper method that takes `XmlReader` and return a `MemberDocumentNode`? This helper will first gather all necessary details (e.g. `summary`, `parameters`, `member`, ...) before instantiating the `MemberDocumentNode` object (we do not want to have more than one constructor for `MemberDocumentNode`, if possible).